lib/my_array.rb

```diff
@@ -0,0 +1,50 @@
+class MyArray < Array
+  def initialize(array)
+    @array = array
+  end
```

Comment:
Is this necessary? Can you not just use the implicit `self` to do things like `drop(1)` without referencing state explicitly?